### DISCUSSION TITLE
Issues getting client-read-h264 example working

### DISCUSSION THREAD

**Discussion by raybjork** [bluenviron/gortsplib#135] [Questions]
First off, this library is awesome thanks for developing and maintaining it!

I have been trying to get the `client-read-h264` demo up and running on my machine and am running into an issue finding the decoder from the avcodec library (`C.avcode_find_decoder(C.AV_CODEC_ID_H264)` is returning nil).  I have followed the installation instructions suggested in the comment of `main.go` and am getting the program to compile, but I am wondering if there is an extra step I need to do to link the library.  

Any advice would be greatly appreciated, and apologies if this is not the correct format for this type of question.

**Comment by aler9** [comment]
Hello, if `avcode_find_decoder` returns nil, it means that the H264 decoder is not installed correctly. Please make sure that
* you're using Debian buster or onwards, or Ubuntu 20.04 or onwards. I've tested instructions on both of them (`apt install -y libavformat-dev libswscale-dev gcc pkg-config`) and found no problems at all 
* Alternatively, make sure that the H264 decoder is installed properly
* On macOS, ome user suggests to add `avcodec_register_all()` before `avcodec_find_decoder()`, it may work for you too https://stackoverflow.com/questions/4091340/ffmpeg-compilation-problem-avcodec-find-decoder-always-returns-null
